src/scipp/io/hdf5.py

```diff
@@ -197,7 +198,8 @@ def read(cls, group):
         values = group['values']
         contents = {key: values.attrs[key] for key in ['dims', 'shape']}
         contents['dtype'] = cls._dtypes[values.attrs['dtype']]
-        contents['unit'] = sc.Unit(values.attrs['unit'])
+        if 'unit' in values.attrs:
+            contents['unit'] = sc.Unit(values.attrs['unit'])
```

Comment:
Hmm, is this wrong actually? Above we store no attribute if the unit is none, but if we do not specify `unit` when calling `empty` it will use `default_unit`?

Reply:
Right. So `contents['unit'] = None` if there is no attr?
Or better store an attr `'none'`.

Reply:
I don't think I want to store `'none'` as a string, since we store units as strings. See the update I had pushed, that seems to work.